test/e2e/functional/same-input-output-path-optional.yaml

```diff
@@ -0,0 +1,33 @@
+apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: optional-input-output-artifact-same-location-
+  labels:
+    argo-e2e: true
+spec:
+  entrypoint: plan
+  templates:
+    - name: plan
+      steps:
+        - - name: write
+            template: file
+        - - name: read
+            template: file
+            arguments:
+              artifacts:
+                - name: file
+                  from: "{{steps.write.outputs.artifacts.file}}"
+    - name: file
+      inputs:
+        artifacts:
+          - name: file
+            path: /tmp/file1.txt
+            optional: true
```

Comment:
I'm not sure optional actually works - can you run without it?

Reply:
Running without `optional: true` fails the validation as expected:

```
$ argo submit art.yaml
2020/05/13 14:56:52 Failed to submit workflow: templates.plan.steps[0].write templates.file inputs.artifacts.file was not supplied
```